scim2-sdk-common/src/main/java/com/unboundid/scim2/common/utils/Parser.java

```diff
@@ -161,6 +177,76 @@ public static Filter parseFilter(final String filterString)
     return readFilter(new StringReader(filterString.trim()), false);
   }
 
+  /**
+   * Add parser options, in order to modify parser behavior within the current
+   * thread.
+   *
+   * <p>NOTE: SCIM server implementations are not guaranteed to support a given option.</p>
+   * <p>NOTE: these should be removed when no longer needed.</p>
+   *
+   * <pre>
+   *   try
+   *   {
+   *     Parser.addOptions(ParserOption.ALLOW_SEMICOLONS_IN_ATTRIBUTE_NAMES();
+   *     performWhateverProcessing();
+   *   }
+   *   finally
+   *   {
+   *     Parser.removeOptions(ParserOption.ALLOW_SEMICOLONS_IN_ATTRIBUTE_NAMES();
```

Comment:
The method should instead call setOptions with whatever the options were before addOptions was called. This guards against the ParserOption.ALLOW_SEMICOLONS_IN_ATTRIBUTE_NAMES option already being set.

Reply:
So the following?

```
Set<ParserOption> priorOptions = null;
try
{
  priorOptions = Parser.addOptions(ParserOption.ALLOW_SEMICOLONS_IN_ATTRIBUTE_NAMES);
  performWhateverProcessing();
}
finally
{
  if (null != priorOptions)
  {
    Parser.setOptions(priorOptions);
  }
}
```